### PR TITLE
Dues transactions in Nova

### LINKED ISSUE #196: Add Graceful Fallback for JS disabled browsers
If JS is disabled, display a message instructing the user to enable JS.

### LINKED ISSUE #352: Condense Dues/Accept Dues, Swag/Give Swag Pages
There are presently four distinct pages for Dues and Swag Ops - Dues, Accept Dues, Swag, Give Swag. This is kinda dumb since all four are just tables with different data. These pages should be collapsed into two - one "Dues" and one "Swag" - with toggle buttons to change the data in the table appropriately and the extraneous menu options removed.

**Dues**
Initial page load should match the as-is page. Add a button to filter the data table entries down to just those who have not paid.

**Swag**
Initial page load should match the as-is page. Add a button to filter the data table entries down to just those who have outstanding swag (needs to be picked up).

### LINKED ISSUE #357: Add Recorded By field to Show Payment component
This was deferred because show payment doesn't currently have access to the name associated with the payment, only the `User_ID`.

A field should be added to the Show Payments component to display who accepted the payment and recorded it into Apiary.

### LINKED ISSUE #439: Dues collection in nova
Remaining work:
 - [x] Action to add payment
 - [x] Action to mark polo picked up
 - [x] Action to mark t-shirt picked up
 - [x] Lens for distributing swag (filter by team, detailed status via eloquent?)
 - [x] Dues transaction filter by team 
 - [x] Sortable fields for transactions
 - [x] Sortable fields for payments
 - [x] Move top level fields on resources out of groups (so action buttons appear on detail pages)

### LINKED ISSUE #509: Disallow admin-entering a payment for yourself
Square checkout payments are ok

### PR DESCRIPTION
**Pull request by kberzinch**
Please sanity check the below issues as well as the code/functionality/etc.

Closes #439 
Closes #196 (DTs can be created by an admin now)
Closes #352 
Closes #357 
Closes #509 

# Todos
 - [x] Permissions don't allow PMs access to nova or viewing users
 - [x] Check transaction fee logic
 - [x] Editing dues transactions and distributing swag are the same permission - let's not do that
 - [x] Actions broken - getAuthIdentifier
 - [x] #569

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+root = true
```
Comment (JoshOldenburg) [inline .editorconfig]: Should this file be in git?

File: app/Nova/Actions/AddPayment.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     */
+    public function handle(ActionFields $fields, Collection $models)
+    {
+        if (count($models) > 1) {
+            return Action::danger(
+                'Payments cannot be entered for more than one transaction at a time. Contact the developers if you think this is unreasonable.'
+            );
+        }
+        // shouldn't happen but might if someone is abusing the API
+        if (Auth::user()->cant('create-payments-'.$fields->method)) {
```
Comment (JoshOldenburg) [inline app/Nova/Actions/AddPayment.php]: Is this injection safe?

File: app/Nova/DuesTransaction.php [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    /**
+     * Get the filters available for the resource.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\DuesTransactionTeam,
```
Comment (JoshOldenburg) [inline app/Nova/DuesTransaction.php]: Should require `read-teams-membership`

File: app/User.php [hunk 4]
```diff
@@ -173,12 +187,12 @@ public function routeNotificationForMail()
...
     public function getAuthIdentifierName()
     {
-        return 'uid';
+        return 'id';
     }
 
     public function getAuthIdentifier()
     {
-        return $this->uid;
+        return $this->id;
```
Comment (JoshOldenburg) [inline app/User.php]: What does this do?

File: .editorconfig (hunk 1 above)
Comment (kberzinch) [reply .editorconfig]: Yes. https://editorconfig.org/

File: app/Nova/Actions/AddPayment.php (hunk 2 above)
Comment (kberzinch) [reply app/Nova/Actions/AddPayment.php]: Probably? @RossIV

File: app/User.php (hunk 4 above)
Comment (kberzinch) [reply app/User.php]: Fixes viewing actions on some resources.

File: app/Nova/DuesTransaction.php (hunk 3 above)
Comment (kberzinch) [reply app/Nova/DuesTransaction.php]: seems pedantic but k

File: app/Nova/DuesTransaction.php [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    public static $with = ['payment', 'package', 'user'];
+
+    /**
+     * Get the displayble label of the resource.
+     *
+     * @return string
+     */
+    public static function label()
+    {
+        return 'Dues Transactions';
```
Comment (JoshOldenburg) [inline app/Nova/DuesTransaction.php]: This shows up as duestransactions in staging

Comment (kberzinch) [reply app/Nova/DuesTransaction.php]: Where are you seeing that?

Comment (JoshOldenburg) [reply app/Nova/DuesTransaction.php]: It was showing up on user detail pages but I can't reproduce it now

Comment (kberzinch) [reply app/Nova/DuesTransaction.php]: Found and fixed

File: app/Nova/Actions/AddPayment.php [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                'You do not have permission to accept that payment method. Please contact a developer.'
+            );
+        }
+
+        if ($fields->method === 'square' || $fields->method === 'swipe') {
+            if ($fields->amount !== ($models->first()->package()->get()->first()->cost) + 3) {
+                $this->markAsFailed($models->first(), null);
+
+                return Action::danger(
+                    'Missing expected transaction fee - total should be '.(($models->first()->package()->get()->first()->cost) + 3).', '.$fields->amount.' entered.'
```
Comment (RossIV) [inline app/Nova/Actions/AddPayment.php]: I'm somewhat opposed to hardcoding the transaction fee.

File: app/Nova/Actions/DistributePolo.php [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+                        if (count($models) === 1) {
+                            return Action::danger('Polo already picked up.');
+                        } else {
+                            $failures[] = $model->id;
+                        }
+                    }
+                } else {
+                    $this->markAsFailed($model, null);
+                    if (count($models) === 1) {
+                        return Action::danger('This transaction is not yet paid.');
```
Comment (RossIV) [inline app/Nova/Actions/DistributePolo.php]: As discussed in Slack, see if we can hide the action if the txn isn't paid.

File: app/Nova/Actions/DistributeShirt.php [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+                        if (count($models) === 1) {
+                            return Action::danger('T-shirt already picked up.');
+                        } else {
+                            $failures[] = $model->id;
+                        }
+                    }
+                } else {
+                    $this->markAsFailed($model, null);
+                    if (count($models) === 1) {
+                        return Action::danger('This transaction is not yet paid.');
```
Comment (RossIV) [inline app/Nova/Actions/DistributeShirt.php]: As discussed in Slack, see if we can hide the action if the txn isn't paid.

File: resources/views/layouts/app.blade.php [hunk 9]
```diff
@@ -23,6 +23,12 @@
               <a class="nav-link" href="{{ route('teams.index') }}">Teams</a>
             </li>
 
+            @can('access-nova')
+            <li class="nav-item">
+              <a class="nav-link" href="/nova/">Nova</a>
```
Comment (RossIV) [inline resources/views/layouts/app.blade.php]: Are we calling this Nova to normal people?

File: app/Nova/Actions/AddPayment.php (hunk 6 above)
Comment (kberzinch) [reply app/Nova/Actions/AddPayment.php]: Do you have an alternative proposal?

File: resources/views/layouts/app.blade.php (hunk 9 above)
Comment (kberzinch) [reply resources/views/layouts/app.blade.php]: It already says Nova a few places in the UI.

I'm open to other suggestions

File: app/Nova/Actions/DistributePolo.php (hunk 7 above)
Comment (kberzinch) [reply app/Nova/Actions/DistributePolo.php]: Done

File: app/Nova/Actions/DistributeShirt.php (hunk 8 above)
Comment (kberzinch) [reply app/Nova/Actions/DistributeShirt.php]: Done

File: app/Nova/Actions/AddPayment.php (hunk 6 above)
Comment (RossIV) [reply app/Nova/Actions/AddPayment.php]: You could put something in .env though that still seems kinda wrong. Approving for now with the requirement that it be revisited later if we ever think of something better.

Comment (RossIV) [reply app/Nova/Actions/AddPayment.php]: Follow up - `5.00` != `5`, per Slack.

File: app/Nova/Actions/AddPayment.php [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+        return [
+            Select::make('Payment Method', 'method')
+                ->options($allowed_payment_methods)
+                ->displayUsingLabels()
+                ->creationRules('required'),
+
+            Currency::make('Amount')
+                ->format('%.2n')
+                ->help('Record the actual amount of money being collected, including surcharges or processing fees.')
```
Comment (RossIV) [inline app/Nova/Actions/AddPayment.php]: Add specific amount of processing fee here.

**Pull request merged**